### PR TITLE
[16.0][ADD] product_packaging_level_sellable

### PR DESCRIPTION
**Pull request by QuocDuong1306**
This module allows odoo users to set:

* Can be sold: On product packaging type model, this checkbox defines if product
  packagings from this particular type are available to be selected on sale
  order line.

### CODE REVIEW

File: product_packaging_level_sellable/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2020 Camptocamp SA
```
Comment (nilshamerlinck) [inline product_packaging_level_sellable/__manifest__.py]: 2023

File: product_packaging_level_sellable/models/product_packaging.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2020 Camptocamp SA
```
Comment (nilshamerlinck) [inline product_packaging_level_sellable/models/product_packaging.py]: 2023

File: product_packaging_level_sellable/models/sale_order_line.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+from odoo.tools import float_compare
+
+
+class SaleOrderLine(models.Model):
+
```
Comment (nilshamerlinck) [inline product_packaging_level_sellable/models/sale_order_line.py]:
```suggestion
```

File: product_packaging_level_sellable/readme/CONFIGURE.rst [hunk 4]
```diff
@@ -0,0 +1,5 @@
+This module allows odoo users to set:
+
+* Can be sold: On product packaging type model, this checkbox defines if product
+  packagings from this particular type are available to be selected on sale
```
Comment (nilshamerlinck) [inline product_packaging_level_sellable/readme/CONFIGURE.rst]:
```suggestion
* Can be sold: On product packaging level model, this checkbox defines if product
  packagings from this particular level are available to be selected on sale
```

File: product_packaging_level_sellable/tests/common.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    @classmethod
+    def setUpClassPartner(cls):
+        cls.partner = cls.env.ref("base.res_partner_12")
+
+    @classmethod
+    def setUpClassProduct(cls):
+        cls.product = cls.env.ref("product.product_product_9")
+
+    @classmethod
+    def setUpClassPackagingType(cls):
```
Comment (nilshamerlinck) [inline product_packaging_level_sellable/tests/common.py]:
```suggestion
    def setUpClassPackagingLevel(cls):
```

File: product_packaging_level_sellable/tests/test_packaging_level_can_be_sold.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        """Check that a product.packaging can be independently set as can be sold."""
+        exception_msg = (
+            "Packaging Test packaging cannot be sold on product {} must be set "
+            "as 'Can be sold' in order to be used on a sale order."
+        ).format(self.product.name)
+        with self.assertRaisesRegex(ValidationError, exception_msg):
+            self.order_line.write(
+                {"product_packaging_id": self.packaging_cannot_be_sold.id}
+            )
+        # Packaging can be sold even if the packaging type does not allows it
```
Comment (nilshamerlinck) [inline product_packaging_level_sellable/tests/test_packaging_level_can_be_sold.py]:
```suggestion
        # Packaging can be sold even if the packaging level does not allows it
```

**Review by nilshamerlinck — CHANGES_REQUESTED** [review]
I think you should also add a new column can be sold in the tree view

**Comment by QuocDuong1306** [comment]
Hi @nilshamerlinck , I updated

**Review by nilshamerlinck — APPROVED** [review]

**Pull request merged**